File: splearn-java/src/main/java/io/wisoft/splearn/application/MemberQueryService.java
package io.wisoft.splearn.application;

import io.wisoft.splearn.application.provided.MemberFinder;
import io.wisoft.splearn.application.required.MemberRepository;
import io.wisoft.splearn.domain.Member;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.validation.annotation.Validated;

@Service
@Transactional
@Validated
@RequiredArgsConstructor
public class MemberQueryService implements MemberFinder {
    private final MemberRepository memberRepository;

    @Override
    public Member find(Long memberId) {
        return memberRepository.findById(memberId).orElseThrow(() -> new IllegalArgumentException("회원을 찾을 수 엇습니다. id: " + memberId));
    }
}

File: splearn-typescript/src/main/application/member-query.service.ts
import { Inject, Injectable } from "@nestjs/common";

import { Member } from "@src/main/domain/member";
import { IllegalArgumentException } from "@src/common/exception/exceptions";

import { MEMBER_REPOSITORY } from "@src/app.token";

import type { MemberFinder } from "@src/main/application/provided/member-finder";
import type { MemberRepository } from "@src/main/application/required/member.repository";

@Injectable()
export class MemberQueryService implements MemberFinder {
  constructor(
    @Inject(MEMBER_REPOSITORY)
    private readonly memberRepository: MemberRepository,
  ) {}


  async find(memberId: number): Promise<Member> {
    const member = await this.memberRepository.findById(memberId);

    if (!member) {
      throw new IllegalArgumentException(`회원을 찾을 수 없습니다. id: ${ memberId }`);
    }

    return member;
  }
}